src/schedule/schedule_dataflow_rewrite.cc

```diff
@@ -182,9 +182,32 @@ Tensor CacheWriteWithReLayout(Schedule sch,
       vsub[iv->var.get()] = value_map.at(iv);
     }
   }
-  Expr body = VarReplacer(vsub).Mutate(compute->body[tensor->value_index]);
-  body = InjectPredicate(predicates, body);
-  body = VarReplacer(vsub2newvar).Mutate(body);
+
+  Expr body;
+  Array<Expr> body_list;
+  const ir::Reduce* first_reduce;
+  bool has_reduce = false;
+  for (auto cbody : compute->body) {
+    body = VarReplacer(vsub).Mutate(cbody);
+    body = InjectPredicate(predicates, body);
+    body = VarReplacer(vsub2newvar).Mutate(body);
+    // Reduce nodes in ONE computeOp must be the same except value_index
```

Comment:
Add CHECKS to validate this, it is a good habit to always validate invariance to avoid bugs in the code.

Reply:
I have update code and add some checks as you said, but I don't know how to validate invariance here, can you give me a example?

Reply:
https://github.com/dmlc/tvm/blob/master/src/op/compute_op.cc#L30